plugins/woocommerce-admin/client/marketing/hooks/useNewCampaignTypes.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from '~/marketing/data-multichannel/constants';
+import {
+	CampaignTypesState,
+	CampaignType as APICampaignType,
+	ApiFetchError,
+} from '~/marketing/data-multichannel/types';
+import { CampaignType } from '~/marketing/types/CampaignType';
+
+type UseNewCampaignTypes = {
+	loading: boolean;
+	data?: Array< CampaignType >;
+	error?: ApiFetchError;
+};
+
+export const useNewCampaignTypes = (): UseNewCampaignTypes => {
+	const convert = ( campaignType: APICampaignType ): CampaignType => {
+		return {
+			id: campaignType.id,
+			icon: campaignType.icon_url,
+			name: campaignType.name,
+			description: campaignType.description,
+			createUrl: campaignType.create_url,
+			channelName: campaignType.channel.name,
+			channelSlug: campaignType.channel.slug,
+		};
+	};
+
+	return useSelect( ( select ) => {
+		const { hasFinishedResolution, getCampaignTypes } = select( STORE_KEY );
+		const campaignTypesState = getCampaignTypes< CampaignTypesState >();
+
+		return {
+			loading: ! hasFinishedResolution( 'getCampaignTypes' ),
+			data: campaignTypesState.data?.map( convert ),
+			error: campaignTypesState.error,
+		};
+	} );
```

Comment:
It would be a bit nicer to specify `deps` for `useSelect` in order not to return new references in every rerendering.

```suggestion
	}, [] );
```

Reply:
Yeah, we should always specify `deps`, I missed that. I fixed it in [`3ba492f` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/3ba492f32b52e39650fc9a5d021279c2afc19689) (moving the `convert` function outside of the hook so it doesn't become a dependency) and in [`39ce470` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/39ce470fe8a3ed7370a46f0815307f7c2d6305da) (specifying empty dependency).

On a related note, ESLint should have caught these dependency issues, I don't know why it is not working for `useSelect`. It does work with normal React hooks like `useEffect`. Do you happen to know why?

Reply:
> Do you happen to know why?

Seems it couldn't distinguish whether a custom React hook is designed to use `deps` or not. Usually it needs to be specified separately. Ref: https://github.com/woocommerce/google-listings-and-ads/pull/780

Reply:
Ahh got it! I think we should contribute it upstream to https://www.npmjs.com/package/@wordpress/eslint-plugin and https://www.npmjs.com/package/@woocommerce/eslint-plugin. I'll see if I can make it happen, maybe after this project. That should help fix or prevent a lot of these errors.

Reply:
> I think we should contribute it upstream to [npmjs.com/package/@wordpress/eslint-plugin](https://www.npmjs.com/package/@wordpress/eslint-plugin)

I have created issue https://github.com/WordPress/gutenberg/issues/49897 and PR https://github.com/WordPress/gutenberg/pull/49900.